File: Android/app/src/main/java/com/simplex/coin2credit/AppDatabase.java
package com.simplex.coin2credit;

import android.arch.persistence.room.Database;
import android.arch.persistence.room.Room;
import android.arch.persistence.room.RoomDatabase;
import android.content.Context;

import com.commonsware.cwac.saferoom.SafeHelperFactory;

@Database(entities = {Transaction.class}, version = 1, exportSchema = false)
public abstract class AppDatabase extends RoomDatabase{
    private static AppDatabase INSTANCE;
    public abstract TransactionDao transactionDao();

    public static AppDatabase getAppDatabase(Context context, char[] password) {
        if(INSTANCE == null) {
            SafeHelperFactory factory= new SafeHelperFactory(password);

            INSTANCE = Room.databaseBuilder(context.getApplicationContext(), AppDatabase.class, "transaction_database")
                    .openHelperFactory(factory)
                    .build();
        }

        return INSTANCE;
    }

    public static void destroyDatabase() {
        INSTANCE = null;
    }
}

File: Android/app/src/main/java/com/simplex/coin2credit/database/AppDatabase.kt
package com.simplex.coin2credit.database

import androidx.room.Database
import androidx.room.Room
import androidx.room.RoomDatabase
import android.content.Context

import com.commonsware.cwac.saferoom.SafeHelperFactory

@Database(entities = [Transaction::class], version = 1, exportSchema = false)
abstract class AppDatabase : RoomDatabase() {
    abstract fun transactionDao(): TransactionDao

    companion object {
        private var INSTANCE: AppDatabase? = null

        fun getAppDatabase(context: Context, password: CharArray): AppDatabase = INSTANCE ?: {
            val db = Room.databaseBuilder(context.applicationContext, AppDatabase::class.java, "transaction_database")
                    .openHelperFactory(SafeHelperFactory(password))
                    .build()

            INSTANCE = db
            db
        }()

        fun destroyDatabase() {
            INSTANCE = null
        }
    }
}
